### PR TITLE
RW-31 Homepage

### PR DESCRIPTION
**Pull request by left23**
- update base theme to latest version - run `composer install`
- https://trello.com/c/VSK2QPMt/137-homepage-headlines
- https://trello.com/c/HiOkMJwa/149-homepage-announcement
- https://trello.com/c/8HJjuEwJ/147-homepage-latest-disasters
- https://trello.com/c/fcWzc64x/148-homepage-latest-blog-post
- https://trello.com/c/LBBYTd3u/158-homepage-opportunities
- component for rw-announcement, rw-article, rw-country, rw-disasters, rw-headlines, rw-homepage, rw-meta, rw-most-read, rw-opportunities, rw-view-more
- extend CD components for cd-block-title, cd-button, cd-card, cd-tag
- update sass-lint git hook to use stylelint
- add twig block to custom module templates to attach libraries in sub theme template overrides, and twig block around html elements that need extra classes so sub theme template overrides can be use twig embed
- Add `clean_class` and `clean_id` twig filters where needed to change underscores for spaces to hyphens
- Add favicons
- Update selectors throughout to reflect CD `#main-content` and D8 `.path-frontpage`
- Add `page-title` block template override so page title is no visible on homepage

### CODE REVIEW

File: html/themes/custom/common_design_subtheme/components/rw-blog/rw-blog.css [hunk 1]
```diff
@@ -0,0 +1,93 @@
+/**
+ * Blog post.
+ *
+ * Latest blog post.
+ *
+ * Id: #blog-post
```
Comment (left23) [inline html/themes/custom/common_design_subtheme/components/rw-blog/rw-blog.css:6]: This ID has changed. It's now `blog` for the Section ID, and `blog_post` for children

Eg. `<article class="rw-river-article rw-river-article--blog_post">`

File: html/themes/custom/common_design_subtheme/components/rw-most-read/rw-most-read.css [hunk 2]
```diff
@@ -0,0 +1,31 @@
+/**
+ * Updates.
+ *
+ * Latest updates or most read content in the past 24 hours.
+ *
+ * Id: #most-read
```
Comment (left23) [inline html/themes/custom/common_design_subtheme/components/rw-most-read/rw-most-read.css:6]: This ID has changed. It's now `most_read` 
Eg. `<section id="most_read" class="rw-river rw-river--most_read">`

Related to https://github.com/UN-OCHA/rwint9-site/pull/16/files#r647354179

**Review by orakili — APPROVED** [review]
@left23 Thanks. Looks good as a first iteration.

I've created some tickets to keep track of some of the things to discuss/modify like the headline images, "view all disasters" link etc.: RW-35 to RW-43.

Merging it.

**Review by orakili — APPROVED** [review]

File: html/themes/custom/common_design_subtheme/components/rw-disasters/rw-disasters.css [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+  position: absolute;
+  top: 50%;
+  left: 6px;
+  overflow: hidden;
+  width: 24px;
+  height: 24px;
+  margin-top: -12px;
+  content: "";
+  background: url(../../img/icons/disaster-icons-sprite.png) -64px 0 no-repeat;
+  background: rgba(0, 0, 0, 0) url(../../img/icons/disaster-icons-sprite.svg) -64px 0 no-repeat;
```
[the remark this answers is not in the document]
Comment (left23) [reply html/themes/custom/common_design_subtheme/components/rw-disasters/rw-disasters.css:48]: Ok, good idea.

File: html/themes/custom/common_design_subtheme/sass/cd/cd-header/_cd-user-menu.scss [hunk 4]
```diff
@@ -19,3 +31,11 @@
 .cd-global-header .menu > li button {
   @extend .cd-user-menu__item;
 }
+
+.cd-global-header .menu {
+  @extend .cd-user-menu;
+
+  // If there's no language switcher in the Global Header, uncomment this line
+  // to apply special treatment to the last link in the list.
+  @extend .cd-user-menu--no-language-switcher;
```
[the remark this answers is not in the document]
Comment (left23) [reply html/themes/custom/common_design_subtheme/sass/cd/cd-header/_cd-user-menu.scss:40]: Ok, noted. Once a language switcher is in place, it should be visually obvious that there are adjustments needed to the login menu as the separator will be missing.

**Pull request merged**